### DISCUSSION TITLE
Deleted My Files by Applying a stash

### DISCUSSION THREAD

**Discussion by Vinyzu** [community/community#32438] [Other Feature Feedback, Questions, & Ideas]
Hello, i had an update i was working one for an week or so,
buti had an open pull before push.
So i pulled it (with Github Desktop) and the changes got into the stash (note that the pull code, not my coded update was in the stash). 
I mistankly applied the stash, and now the stash is there, but my update code is gone.
I tried reverting the stash, didnt worked, git says no stash is there.
I cant find my code in receicle bin, its just gone.
How can i reverse the stash/revive my code?

**Comment by Samsuhhh** [comment]
I think the best way to handle this issue is by using the "**git reflog** coupled with either the "git reset or git revert" command. By using git reflog, you can pull up a log (history) of the changes made to the HEAD, which are differentiated by reference id numbers. There are two ways to go about this: 

- git reset --HARD <ref number>
(this method will basically reset the history back to the point you input; erasing all commits after input point)

- git revert <ref number>
(this method makes a new commit, undoing all the commits up the point you specify)

To execute these fixes, you start the same way by using the **git reflog** command to bring up the history of changes to the HEAD.
1. Identify a version of your code from before your pull request updated the code. 
2. Copy or save the reference number for a version BEFORE your unwanted stash (ref number is a string of numbers). 

3a. Use **git revert <ref number>** to revert the HEAD back to the ref number you specified. 
       ie: You have a ref number 123, but your head is currently 5 commits ahead of it. Once you use revert, it will make a 6th commit 
          that will update your code to reflect whatever ref number you specified. 
OR 

3b. Use **git reset --HARD <ref number>** to reset the HEAD back in time to the ref number you specified. 
(**NOTE: This method will DELETE all commits made after the point in time of the ref number specified.**)
       ie: You want to go back to the ref number 123, but your head is currently 5 commits ahead of it. Once you use reset --HARD, all 
         5 commits will be erased, bringing you back in time to the ref number specified.

**Reply by Vinyzu** [reply]
The problem is, i didnt pushed the updates before. Their not on any server, if they still exist, theyre on my machine
